File: tests/fixtures/amqplib/basic/producer.ts
import amqp from 'amqplib';

export async function publishOrder() {
  const conn = await amqp.connect('amqp://localhost');
  const channel = await conn.createChannel();
  channel.publish('orders', 'order.created', Buffer.from('data'));
}

export async function sendToQueueDirect() {
  const conn = await amqp.connect('amqp://localhost');
  const channel = await conn.createChannel();
  channel.sendToQueue('emails', Buffer.from('data'));
}

export async function dynamicTopic(routingKey: string) {
  const conn = await amqp.connect('amqp://localhost');
  const channel = await conn.createChannel();
  channel.publish('orders', routingKey, Buffer.from('data'));
}

File: tests/fixtures/pika/basic/producer.py
"""pika producer surface."""

import pika


def publish_order():
    conn = pika.BlockingConnection()
    channel = conn.channel()
    channel.basic_publish(exchange='orders', routing_key='order.created', body=b'data')


def publish_audit():
    conn = pika.BlockingConnection()
    channel = conn.channel()
    channel.basic_publish(exchange='audit', routing_key='audit.write', body=b'data')


def dynamic_topic(routing_key: str):
    conn = pika.BlockingConnection()
    channel = conn.channel()
    channel.basic_publish(exchange='orders', routing_key=routing_key, body=b'data')
